subsys/pm/pm.c

```diff
@@ -107,26 +136,40 @@ static void pm_resume_devices(void)
 #endif  /* !CONFIG_PM_DEVICE_RUNTIME_EXCLUSIVE */
 #endif	/* CONFIG_PM_DEVICE */
 
+/* Retrieve the element index of a power state. */
+uint8_t pm_state_get_index(enum pm_state state, uint8_t substate_id)
+{
+	for (uint8_t i = 0; i < ARRAY_SIZE(pmstates); i++) {
+		if (pmstates[i].state == state &&
+		    pmstates[i].substate_id == substate_id) {
+			return i;
+		}
+	}
+
+	return -1;
+}
+
 /*
  * Function called to notify when the system is entering / exiting a
  * power state
  */
-static inline void pm_state_notify(bool entering_state)
+static void pm_state_notify(uint8_t direction)
 {
 	struct pm_notifier *notifier;
 	k_spinlock_key_t pm_notifier_key;
-	void (*callback)(enum pm_state state);
+	uint8_t id = CURRENT_CPU;
```

Comment:
leftover ?

Reply:
The `id` var? I use that on the line below to retrieve the current system state. I thought it was a bit more readable that way (at the cost of a byte of stack mem).